python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -136,33 +138,41 @@ def from_file_discovery_and_command_line_config(
         path: Path,
         command_line_config: DgRawCliConfig,
     ) -> Self:
-        config_path = discover_config_file(path)
+        root_config_path = discover_config_file(path)
         workspace_config_path = discover_config_file(
             path, lambda x: bool(x.get("directory_type") == "workspace")
         )
 
-        if config_path:
-            root_path = config_path.parent
-            root_file_config = load_dg_root_file_config(config_path)
-            if workspace_config_path:
-                workspace_root_path = workspace_config_path.parent
-
-                # If the workspace config is different from the root config, we need to load the
-                # workspace config.
-                container_workspace_file_config = (
-                    load_dg_workspace_file_config(workspace_config_path)
-                    if config_path != workspace_config_path
-                    else None
-                )
+        if root_config_path:
+            root_path = root_config_path.parent
+            root_file_config = load_dg_root_file_config(root_config_path)
+            if workspace_config_path is None:
+                workspace_root_path = None
+                container_workspace_file_config = None
 
-            else:
+            # Only load the workspace config if the workspace root is different from the first
+            # detected root.
+            elif workspace_config_path == root_config_path:
+                workspace_root_path = workspace_config_path.parent
                 container_workspace_file_config = None
-                workspace_root_path = None
+            else:
+                workspace_root_path = workspace_config_path.parent
+                container_workspace_file_config = load_dg_workspace_file_config(
+                    workspace_config_path
+                )
+                if "cli" in root_file_config:
+                    del root_file_config["cli"]
+                    warnings.warn(
+                        generate_tool_dg_cli_in_project_in_workspace_error_message(
+                            root_path, workspace_root_path
+                        )
```

Comment:
how intentional was choosing warnings.warn vs. logger.warn on some click logger or something? i've seen some weird formatting on warnings.warn calls, not sure if applicable here https://dagsterlabs.slack.com/archives/C03A0D72A6T/p1717450735784099

Reply:
Not intentional at all, if you think a different output API would be better I'm happy to change it.